smart-contract/assembly/contracts/deweb-interface.ts

```diff
@@ -1,187 +1,369 @@
 import {
   balance,
   Context,
+  generateEvent,
+  setBytecode,
   sha256,
   transferCoins,
 } from '@massalabs/massa-as-sdk';
 import {
   _onlyOwner,
   _setOwner,
 } from '@massalabs/sc-standards/assembly/contracts/utils/ownership-internal';
-import {
-  ChunkPost,
-  ChunkGet,
-  PreStore,
-  ChunkDelete,
-} from './serializable/Chunk';
 
 import { Args, stringToBytes, u32ToBytes } from '@massalabs/as-types';
+import { FileChunkPost } from './serializable/FileChunkPost';
+import { FileDelete } from './serializable/FileDelete';
+import { FileChunkGet } from './serializable/FileChunkGet';
+import { Metadata } from './serializable/Metadata';
+import {
+  _editFileMetadata,
+  _editGlobalMetadata,
+  _getFileMetadata,
+  _getGlobalMetadata,
+  _removeFileMetadata,
+  _removeGlobalMetadata,
+} from './internals/metadata';
 import {
+  _setFileChunk,
   _getFileChunk,
   _getTotalChunk,
-  _setFileChunk,
-  _setTotalChunk,
   _deleteFile,
   _removeChunksRange,
-  _removeFile,
 } from './internals/chunks';
-
-import { _removeFilePath } from './internals/file-list';
-import { FILES_PATH_LIST } from './internals/const';
-import { _pushFilePath } from './internals/file-list';
+import { _getFileLocations } from './internals/location';
+import { _fileInit } from './internals/fileInit';
+import { FileInit } from './serializable/FileInit';
 
 /**
  * Initializes the smart contract.
- * Sets the contract deployer as the owner and initializes an empty file path list.
- * @param _ - Unused parameter (required).
+ * Sets the contract deployer as the owner.
  */
 export function constructor(_: StaticArray<u8>): void {
   if (!Context.isDeployingContract()) return;
+
```

Comment:
maybe set the deweb version tag here. See https://forum.massa.community/t/community-feedback-proposal-for-improving-website-storage-on-deweb/82/15#p-319-smart-contract-datastore-layout-1

Reply:
to remove any ambiguity with other products, the tag should contain the explicit string "deweb". That way we can unambuously auto-detect that an address is hosting a website and under which version of the format

Reply:
for example it could be [VERSION_TAG][string: "massa_website"] -> "1"